### PR TITLE
Gui/selection manager changes

### PR DESCRIPTION
**Pull request by justinvdk**
This branch cleaned up the SelectionManager a bit.

It now only holds a single selection, either something from the graph or something from the tree.
The searchPaneController can then listen and show the currently selected item.

@FarisElghlan came up with a cool feature, of which a basis is built into this branch. With it you can pin / copy over your selection to a secondary selection buffer. For now all that it can do is show the secondary selection but the plan is to be able to compare the two.

### CODE REVIEW

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/graph/ViewGraphNodeRectangle.java [hunk 1]
```diff
@@ -5,8 +5,10 @@
 import javafx.scene.Node;
 import javafx.scene.paint.Color;
 import javafx.scene.shape.Rectangle;
+import nl.tudelft.pl2016gr2.gui.view.selection.GraphBubbleDescription;
+import nl.tudelft.pl2016gr2.gui.view.selection.ISelectable;
```
Comment (Pathemeous) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/graph/ViewGraphNodeRectangle.java]: nice import!! :feelsgood:

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/GraphBubbleDescription.java [hunk 2]
```diff
@@ -36,8 +39,6 @@ public Node getNode() {
     }
   }
 
-  public abstract String getText();
-
   public static class GraphNodeRectangleDescriptionController implements Initializable {
```
Comment (cbilstra) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/GraphBubbleDescription.java:50]: Maybe also write javadoc for this class?

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/SelectionPaneController.java [hunk 3]
```diff
@@ -1,56 +1,134 @@
...
+      pane.setMaxWidth(150);
+      expandInAnchorPane(pane, compareSelectionPane);
+    });
+  }
+
+  private void resetCompare() {
+    compareSelectionPane.getChildren().clear();
+    compareSelectionPane.getChildren().add(buttonContainer);
+  }
+
```
Comment (cbilstra) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/SelectionPaneController.java:107]: I think that we agreed on also writing javadoc for private methods.

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/tree/TreePaneController.java [hunk 4]
```diff
@@ -313,22 +316,26 @@ public void setOnLeavesChanged(Observer observer) {
     return currentRoot.getCurrentLeaves();
   }
```
Comment (cbilstra) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/tree/TreePaneController.java:319]: Javadoc?

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/GraphBubbleDescription.java (hunk 2 above)
Comment (justinvdk) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/GraphBubbleDescription.java:50]: Added comment

File: PL2/PL2-gui/src/test/java/nl/tudelft/pl2016gr2/gui/IntegrationTest.java [hunk 5]
```diff
@@ -55,59 +45,60 @@ public void launchApplicationTest() {
...
+  //    AnchorPane rootPane = getSelectionController().rootPane;
+  //    Button compareButton = null;
+  //    for (Node node : rootPane.getChildren()) {
+  //      if (node instanceof Pane) {
+  //        compareButton = (Button) ((Pane) node).getChildren().get(0);
+  //      }
+  //    }
+  //    assertNotEquals(null, compareButton);
+  //    compareButton.fire();
+  //  }
```
Comment (cbilstra) [inline PL2/PL2-gui/src/test/java/nl/tudelft/pl2016gr2/gui/IntegrationTest.java]: Delete the commented code?

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/tree/TreePaneController.java (hunk 4 above)
Comment (justinvdk) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/tree/TreePaneController.java:319]: Added

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/SelectionPaneController.java (hunk 3 above)
Comment (justinvdk) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/SelectionPaneController.java:107]: Added several, do notify when you find more that you'd like to see javadoc for.

File: PL2/PL2-gui/src/test/java/nl/tudelft/pl2016gr2/gui/IntegrationTest.java (hunk 5 above)
Comment (justinvdk) [inline PL2/PL2-gui/src/test/java/nl/tudelft/pl2016gr2/gui/IntegrationTest.java]: Good point, removed.

**Comment by wouterjsmit** [comment]
Sounds good

**Pull request merged**